### PR TITLE
fix the issue that one-layer torch model cannot be trained

### PR DESCRIPTION
**Pull request by geez0219**
1. fix the issue that one-layer torch model cannot be trained
2. add the corresponding testing code

### CODE REVIEW

File: fastestimator/backend/get_gradient.py [hunk 1]
```diff
@@ -77,8 +77,23 @@ def get_gradient(target: Tensor,
...
                                         only_inputs=True)
-        if isinstance(gradients, Tuple) and len(gradients) == 1:
-            # PyTorch seems to unnecessarily wrap results into a tuple when a non-scalar target is provided
+
+        if isinstance(gradients, Tuple) and len(gradients) == 1 and isinstance(sources, torch.Tensor):
+            #  The behavior table of tf and torch backend
+            #  ---------------------------------------------------------------
+            #        | case 1                     | case 2                    |
+            #  ---------------------------------------------------------------|
+            #  tf    | taget: tf.Tensor           | taget: tf.Tensor          |
```
Comment (TortoiseHam) [inline fastestimator/backend/get_gradient.py:86]: 'target' rather than 'taget'

File: fastestimator/backend/get_gradient.py [hunk 2]
```diff
@@ -77,8 +77,23 @@ def get_gradient(target: Tensor,
                                         retain_graph=retain_graph,
                                         create_graph=higher_order,
                                         only_inputs=True)
-        if isinstance(gradients, Tuple) and len(gradients) == 1:
-            # PyTorch seems to unnecessarily wrap results into a tuple when a non-scalar target is provided
+
+        if isinstance(gradients, Tuple) and len(gradients) == 1 and isinstance(sources, torch.Tensor):
```
Comment (TortoiseHam) [inline fastestimator/backend/get_gradient.py:81]: do we still need to check that len(gradients) == 1 if we're checking that sources is a torch.Tensor?

File: fastestimator/backend/get_gradient.py (hunk 1 above)
Comment (geez0219) [reply fastestimator/backend/get_gradient.py:86]: Apparently i have a spelling issue.

File: fastestimator/backend/get_gradient.py (hunk 2 above)
Comment (geez0219) [reply fastestimator/backend/get_gradient.py:81]: I was considering you may see some cases that I didn't, so I just added an additional condition.
Since you also think that is unnecessary then I will simplify it.

Comment (geez0219) [reply fastestimator/backend/get_gradient.py:81]: In that sense, I might also discard "isinstance(gradients, Tuple)" condition

Comment (TortoiseHam) [reply fastestimator/backend/get_gradient.py]: so long as your code comments about possible combinations is exhaustive. have you run pytorch saliency with these changes?

Comment (geez0219) [reply fastestimator/backend/get_gradient.py]: No, but i have run nightly-test and it passed

Comment (geez0219) [reply fastestimator/backend/get_gradient.py]: Just ran the xai pytorch version that was modified from t08_xai.ipynb and it works
